run/ingress/main.tf

```diff
@@ -25,34 +25,26 @@ resource "google_project_service" "cloudrun_api" {
 }
 
 # [START cloudrun_service_ingress]
-resource "google_cloud_run_service" "default" {
+resource "google_cloud_run_v2_service" "default" {
   provider = google-beta
   name     = "ingress-service"
   location = "us-central1"
 
+  # For valid annotation values and descriptions, see
+  # https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_v2_service#ingress
+  ingress = "INGRESS_TRAFFIC_INTERNAL_ONLY"
+
   template {
-    spec {
-      containers {
-        image = "gcr.io/cloudrun/hello" #public image for your service
-      }
+    containers {
+      image = "gcr.io/cloudrun/hello" #public image for your service
     }
   }
-  traffic {
-    percent         = 100
-    latest_revision = true
-  }
-  metadata {
-    annotations = {
-      # For valid annotation values and descriptions, see
-      # https://cloud.google.com/sdk/gcloud/reference/run/deploy#--ingress
-      "run.googleapis.com/ingress" = "internal"
-    }
-  }
-
+  # [END cloudrun_service_ingress]
   lifecycle {
```

Comment:
non-blocking: a note on what lifecycle is and why it's being used would be nice

Reply:
Added context on this and the other example.